flang/lib/Lower/OpenMP.cpp

```diff
@@ -426,14 +425,10 @@ void DataSharingProcessor::collectSymbols(
                              /*collectSymbols=*/true,
                              /*collectHostAssociatedSymbols=*/true);
   for (Fortran::lower::pft::Evaluation &e : eval.getNestedEvaluations()) {
-    if (e.hasNestedEvaluations())
+    if (e.hasNestedEvaluations() && !e.isConstruct())
       converter.collectSymbolSet(e, symbolsInNestedRegions, flag,
                                  /*collectSymbols=*/true,
                                  /*collectHostAssociatedSymbols=*/false);
-    else
-      converter.collectSymbolSet(e, symbolsInParentRegions, flag,
-                                 /*collectSymbols=*/false,
-                                 /*collectHostAssociatedSymbols=*/true);
```

Comment:
With this change and that in `collectSymbolSet()`, it seems that host associated symbols are never collected anymore, isn't it? If this is the case, can't we remove the `collectHostAssociatedSymbols` and `collectSymbols` flags from `collectSymbolSet()`?

Reply:
I guess not, because it's an `if (collectSymbol && oriSymbol.test(flag)`. Sorry for the noise.